import OpenAI from "openai"
import {ImageGenerationFailedException} from "../constants/custom_exceptions"

export const OpenaiService = {

  /**
   * Returns the url of a baboon image generated by AI
   * @throws ImageGenerationFailedException if an error was encountered along the way
   */
  generateBaboonImage: async (): Promise<string> => {
    let result

    try {
      const openai = new OpenAI({
        apiKey: process.env.OPENAI_API_KEY,
        project: process.env.OPENAI_PROJECT_ID
      })

      const response = await openai.images.generate({
        model: "dall-e-3",
        prompt: "A photo realistic image of an ugly baboon",
        n: 1,
        size: "1024x1024",
      })

      result = response.data[0].url
    }
    catch (e) {
      throw new ImageGenerationFailedException()
    }

    if (!result) throw new ImageGenerationFailedException()
    return result
  },

}